colcon_cargo/task/cargo/build.py

```diff
@@ -108,17 +108,23 @@ def _prepare(self, env, additional_hooks):
     # Overridden by colcon-ros-cargo
     def _build_cmd(self, cargo_args):
         args = self.context.args
-        return [
+        cmd = [
             CARGO_EXECUTABLE,
             'build',
             '--quiet',
             '--target-dir', args.build_base,
-        ] + cargo_args
+        ]
+        if not any(
+            arg == '--profile' or arg.startswith('--profile=')
```

Comment:
```suggestion
            arg == '--profile' or arg.startswith('--profile=') or arg == '--release'
```

The `release` profile can be specified with both `--profile release` or just `--release` and (I believe) they do the same.
I personally use `--release` to save typing and I suspect that's what most people would do, in which case the current check wouldn't catch the flag and we would end up doing a `cargo build --release --profile dev` which errors out:

```
$ cargo build --release --profile dev
error: conflicting usage of --profile=dev and --release
The `--release` flag is the same as `--profile=release`.
Remove one flag or the other to continue.
```

Reply:
I don't think it's possible to use `--release` or `--debug` when both 'build' and 'install' are invoked. Unfortunately `--release` isn't an argument to 'install', and `--debug` isn't an argument to 'build'.

Reply:
Ah :| I admit using `colcon-ros-cargo` most so haven't tried this standalone too much, but yea that makes sense.

Reply:
We could consider adding some smarts here to "do the right thing" when the developer specifies `--release` or `--debug` by just changing it to `--profile release` and `--profile dev`, but I'm not sure how I feel about it.

Reply:
Yea I agree with your current approach, if it's an error in `cargo` it should probably be kept that way, rather than trying to change it ourselves